### PR TITLE
[MI-2040]: Removed the hard coded gitlab.com in Terms of Service error

### PR DESCRIPTION
**Pull request by Nityanand13**
Removed the hard coded gitlab.com in Terms of Service error

### CODE REVIEW

File: app/oauth.go [hunk 1]
```diff
@@ -897,8 +897,10 @@ func (a *App) AuthorizeOAuthUser(w http.ResponseWriter, r *http.Request, service
 		mlog.Error("Error getting OAuth user", mlog.Int("response", resp.StatusCode), mlog.String("body_string", bodyString))
 
 		if service == model.ServiceGitlab && resp.StatusCode == http.StatusForbidden && strings.Contains(bodyString, "Terms of Service") {
+			url, _ := url.Parse(*sso.UserAPIEndpoint)
```
Comment (Kshitij-Katiyar) [inline app/oauth.go]: check its error

File: app/oauth.go [hunk 2]
```diff
@@ -897,8 +897,10 @@ func (a *App) AuthorizeOAuthUser(w http.ResponseWriter, r *http.Request, service
...
+			mapURL:=map[string]interface{}{"URL": url.Hostname()}
```
Comment (Kshitij-Katiyar) [inline app/oauth.go]: give better name than mapURL

**Review by Kshitij-Katiyar — APPROVED** [review]

File: app/oauth.go [hunk 3]
```diff
@@ -897,8 +897,13 @@ func (a *App) AuthorizeOAuthUser(w http.ResponseWriter, r *http.Request, service
 		mlog.Error("Error getting OAuth user", mlog.Int("response", resp.StatusCode), mlog.String("body_string", bodyString))
 
 		if service == model.ServiceGitlab && resp.StatusCode == http.StatusForbidden && strings.Contains(bodyString, "Terms of Service") {
+			url, err := url.Parse(*sso.UserAPIEndpoint)
+			if err != nil {
+				return nil, "", stateProps, nil, model.NewAppError("AuthorizeOAuthUser", "api.user.authorize_oauth_user.service.app_error", map[string]any{"Service": service}, err.Error(), http.StatusInternalServerError)
+			}
+			urlParamMap:=map[string]interface{}{"URL": url.Hostname()}
```
Comment (manojmalik20) [inline app/oauth.go]:
```suggestion
```

File: app/oauth.go [hunk 4]
```diff
@@ -897,8 +897,13 @@ func (a *App) AuthorizeOAuthUser(w http.ResponseWriter, r *http.Request, service
...
 			// Return a nicer error when the user hasn't accepted GitLab's terms of service
-			return nil, "", stateProps, nil, model.NewAppError("AuthorizeOAuthUser", "oauth.gitlab.tos.error", nil, "", http.StatusBadRequest)
+			return nil, "", stateProps, nil, model.NewAppError("AuthorizeOAuthUser", "oauth.gitlab.tos.error", urlParamMap, "", http.StatusBadRequest)
```
Comment (manojmalik20) [inline app/oauth.go]:
```suggestion
			return nil, "", stateProps, nil, model.NewAppError("AuthorizeOAuthUser", "oauth.gitlab.tos.error",
			    map[string]interface{}{"URL": url.Hostname()}, "", http.StatusBadRequest)
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**